### PR TITLE
Addded Gulp Zip

### PR DESCRIPTION
**Pull request by NaveenKharwar**
## Description 
📦 NEW: Gulp zip added
( https://www.npmjs.com/package/gulp-zip ) so that user can generate zip folder of there theme and also Gulp Zip will remove unwanted files and build a zip that is ready to use or submit at WordPress Repo.

**Gulp Task** 👇👇

gulp.task('zip', function () {
	return gulp.src([
		'./**/*', '!./{node_modules,node_modules/**/*}',
		'!./.git',
		'!./.svn',
		'!./gulpfile.babel.js', 
		'!./wpgulp.config.js',
		'!./.eslintrc.js',
		'!./.eslintignore',
		'!./.editorconfig',
		'!./phpcs.xml.dist',
		'!./vscode',
		'!./package.json', 
		'!./package-lock.json'])
		.pipe(zip('my-theme-folder.zip'))
		.pipe(gulp.dest('./../'));
  });

## Screenshots: 

👉 https://ibb.co/p1TqJKV

## Types of changes

📦 NEW: Gup Zip Added 

## How Has This Been Tested?

I tested this while building a WordPress theme.

## Checklist:
- [👍 ] My code is tested.
- [👍 ] My code follows the WordPress code style.
- [👍 ] My code follows has extensive inline documentation like the rest of WPGulp.

### CODE REVIEW

File: src/gulpfile.babel.js [hunk 1]
```diff
@@ -33,6 +35,9 @@ const config = require( './wpgulp.config.js' );
  */
 const gulp = require( 'gulp' ); // Gulp of-course.
 
+// Generates zip of theme.
```
Comment (ahmadawais) [inline src/gulpfile.babel.js]: WP Gulp is used by both themes and plugins. Adding something just for themes is not right. Care to test for both and adjust the lingo?

File: src/gulpfile.babel.js [hunk 2]
```diff
@@ -363,3 +368,28 @@ gulp.task(
 		gulp.watch( config.imgSRC, gulp.series( 'images', reload ) ); // Reload on customJS file changes.
 	})
 );
+
+/**
+ * Generate .zip
+ *
+ * Generate my-theme-folder.zip in your `/wp-content/themes` folder.
```
Comment (ahmadawais) [inline src/gulpfile.babel.js]: WP Gulp is used by both themes and plugins. Adding something just for themes is not right. Care to test for both and adjust the lingo?

**Comment by ahmadawais** [comment]
Thanks for this. Can you adjust the tests and docs as requested? Peace! ✌️

**Comment by naveenkharwar** [comment]
Hey @ahmadawais :) this is my first FOSS contribution other than WordPress 

> Can you adjust the tests and docs as requested?

I didn't get this Sorry for silly question!

File: src/gulpfile.babel.js (hunk 2 above)
Comment (naveenkharwar) [reply src/gulpfile.babel.js]: Okay Got it I will test it on Plugin also and then send again pull request.

Comment (ahmadawais) [reply src/gulpfile.babel.js]: Sure, looking forward.

File: src/gulpfile.babel.js (hunk 1 above)
Comment (ahmadawais) [reply src/gulpfile.babel.js]: @naveenkharwar you mention the word `theme` it should be `theme/plugin` since WPGulp works with both.

**Comment by ahmadawais** [comment]
@naveenkharwar welcome here 👍 
Glad to see you interested in open source.

Just go through the code review comments, reply, resolve them once you have made the right edits.

Peace! ✌️

**Comment by bilalmalkoc** [comment]
It must zip inside theme folder. But it doesn't.

**Comment by martinkrcho** [comment]
@ahmadawais , are you planning to merge this branch any time soon? Is there something I can help with?

**Comment by ahmadawais** [comment]
Thank you @NaveenKharwar for the initial work — this needed some more refactoring that I just added in v3 release https://github.com/ahmadawais/WPGulp/issues/188#issuecomment-716127011

**Pull request closed without merging**